pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -52,63 +55,55 @@ protected int getReplication(TableConfig tableConfig) {
   @Override
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
+    SegmentAssignmentStrategy segmentAssignmentStrategy = _assignmentStrategyMap.get(InstancePartitionsType.OFFLINE);
+    // Need to check assignmentStrategy for Dim Tables as following pre conditions state would return false
+    if (segmentAssignmentStrategy instanceof DimTableSegmentAssignmentStrategy) {
+      return segmentAssignmentStrategy.assignSegment(segmentName,
+          currentAssignment, null, InstancePartitionsType.OFFLINE);
+    }
     InstancePartitions instancePartitions = instancePartitionsMap.get(InstancePartitionsType.OFFLINE);
     Preconditions.checkState(instancePartitions != null, "Failed to find OFFLINE instance partitions for table: %s",
         _tableNameWithType);
```

Comment:
```suggestion
    InstancePartitions instancePartitions = instancePartitionsMap.get(InstancePartitionsType.OFFLINE);
    Preconditions.checkState(
        !(segmentAssignmentStrategy instanceof DimTableSegmentAssignmentStrategy) && instancePartitions != null,
        "Failed to find OFFLINE instance partitions for table: %s", _tableNameWithType);
```
One way of rewriting to not bypass the logging below and avoid calling `segmentAssignmentStrategy.reassignSegments` directly.